server/src/endpoints/minidump.rs

```diff
@@ -0,0 +1,161 @@
+fn read_multipart_data(
+    field: multipart::Field<Payload>,
+    max_size: usize,
+) -> ResponseFuture<Vec<u8>, BadStoreRequest> {
+    let future = field.map_err(|_| BadStoreRequest::InvalidMultipart).fold(
+        Vec::with_capacity(MULTIPART_DATA_INITIAL_CHUNK_SIZE),
+        move |mut body, chunk| {
+            if (body.len() + chunk.len()) > max_size {
+                Err(BadStoreRequest::PayloadError(StorePayloadError::Overflow))
+            } else {
+                body.extend_from_slice(&chunk);
+                Ok(body)
+            }
+        },
+    );
+    Box::new(future)
+}
+
+fn handle_multipart_stream(
+    content: SizeLimitedEnvelope,
+    stream: multipart::Multipart<Payload>,
+) -> ResponseFuture<SizeLimitedEnvelope, BadStoreRequest> {
+    let future = stream
+        .map_err(|_| BadStoreRequest::InvalidMultipart)
+        .fold(content, move |content, item| {
+            handle_multipart_item(content, item)
+        });
+    Box::new(future)
+}
+
+fn handle_multipart_item(
+    mut content: SizeLimitedEnvelope,
+    item: multipart::MultipartItem<Payload>,
+) -> ResponseFuture<SizeLimitedEnvelope, BadStoreRequest> {
+    let field = match item {
+        multipart::MultipartItem::Field(field) => field,
+        multipart::MultipartItem::Nested(nested) => {
+            return handle_multipart_stream(content, nested);
+        }
+    };
+
+    let (name, file_name) = field
+        .content_disposition()
+        .as_ref()
+        .map_or((None, None), |d| {
+            (
+                d.get_name().map(String::from),
+                d.get_filename().map(String::from),
+            )
+        });
+
+    if name.is_none() && file_name.is_none() {
+        // log::trace!("multipart content without name or file_name");
+        return Box::new(ok(content));
+    } else {
+        let result = read_multipart_data(field, content.remaining_size).map(|data| {
+            content.remaining_size -= data.len();
+            match (name, file_name) {
+                (name, Some(file_name)) => {
+                    let mut item = Item::new(ItemType::Attachment);
+                    item.set_payload(ContentType::OctetStream, data);
+                    item.set_filename(file_name);
+                    name.map(|name| item.set_header("name", name));
+                    content.envelope.add_item(item)
+                }
+                (Some(name), None) => {
+                    let value = from_utf8(&data);
+                    match value {
+                        Ok(_value) => {
+                            let mut item = Item::new(ItemType::FormData);
+                            item.set_header("name", name);
+                            item.set_payload(ContentType::Text, data);
+                            content.envelope.add_item(item);
+                        }
+                        Err(_failure) => {
+
+                            // log::trace!("invalid text value in multipart item");
```

Comment:
I think we should uncomment those logs as-is, and have another trace() for the `(None, None)` case until we have a better understanding of what is important to log (historically this has been once the service is live)